components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java

```diff
@@ -199,6 +200,54 @@ public void deactivateTenant(String tenantUniqueID) throws TenantMgtException {
         }
     }
 
+    @Override
+    public void deleteTenantMetaData(String tenantUniqueIdentifier) throws TenantMgtException {
+
+        TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
+        if (tenantManager != null) {
+            Tenant tenant = null;
+            try {
+                tenant = tenantManager.getTenant(tenantUniqueIdentifier);
+            } catch (UserStoreException e) {
+                throw new TenantManagementServerException("Error while getting the tenant with the tenant unique id :" +
+                        tenantUniqueIdentifier + " .", e);
+            }
+
+            if (tenant != null) {
+                int tenantId = tenant.getId();
+                String tenantDomain = tenant.getDomain();
+
+                try {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("Starting tenant deletion for domain: %s and tenant id: %d from the " +
+                                "system.", tenantDomain, tenantId));
+                    }
+
+                    ServerConfigurationService
+                            serverConfigurationService = TenantMgtServiceComponent.getServerConfigurationService();
+
+                    if (Boolean.parseBoolean(serverConfigurationService.getFirstProperty("Tenant.TenantDelete"))) {
```

Comment:
at the beginning of the method, we need to do this validation, if this is false, we can just throw.
```
if (!Boolean.parseBoolean(serverConfigurationService.getFirstProperty("Tenant.TenantDelete"))) {
    throw new TenantManagementClientException(ERROR_CODE_TENANT_DELETION_NOT_ENABLED.getCode(),
                                ERROR_CODE_TENANT_DELETION_NOT_ENABLED.getMessage());
}
```